tests/daemon/unit/api/flow_with_env.yml

```diff
@@ -0,0 +1,14 @@
+jtype: Flow
+with:
+  name: my-flow
+  grpc_data_requests: true
+  port_expose: 12345
+  protocol: http
+  env:
+    key1: ${{ context_var1 }}  # this should be replaced
+    key2: $context_var2  # this should not be replaced
+    key3: ${ context_var3 }  # this should not be replaced
```

Comment:
User's Flow yaml might include the first 2 syntaxes. Is there any reason, the 2nd one shouldn't be replaced?

Reply:
it is not supported by `load_config` while testing. maybe I can allow 2nd case happen

Reply:
But the 2nd syntax is available via an environment variable?

Reply:
i created a minimum example:

```python
from jina.jaml import JAML
from pprint import pprint

with open('toy.yml')as fp:
    a = JAML.load(fp)
    b = JAML.expand_dict(a, context={'context_var1': 'MMM', 'context_var2': 'WWW'})
pprint(b)
```

with YAML:
```yaml
jtype: Flow
with:
  name: my-flow
  grpc_data_requests: true
  port_expose: 12345
  protocol: http
  random_key: ${{ context_var1 }}
  random_key2: $context_var2
executors:
  - name: local_replicas
    port_in: 45678
    replicas: 2

```

Reply:
essentially, flow.load_config is calling this `expand_dict` where we pass a `context`